library/src/main/java/com/chuckerteam/chucker/internal/support/FormatUtils.kt

```diff
@@ -0,0 +1,178 @@
+import com.google.gson.JsonParseException
+import com.google.gson.JsonParser
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.PrintWriter
+import java.io.StringWriter
+import java.util.Locale
+import javax.xml.XMLConstants
+import javax.xml.transform.OutputKeys
+import javax.xml.transform.TransformerException
+import javax.xml.transform.sax.SAXSource
+import javax.xml.transform.sax.SAXTransformerFactory
+import javax.xml.transform.stream.StreamResult
+import kotlin.math.ln
+import kotlin.math.pow
+import org.xml.sax.InputSource
+
+internal object FormatUtils {
+
+    private const val SI_MULTIPLE = 1000
+    private const val BASE_TWO_MULTIPLE = 1024
+
+    fun formatHeaders(httpHeaders: List<HttpHeader>?, withMarkup: Boolean): String {
+        return httpHeaders?.joinToString(separator = "") { header ->
+            (if (withMarkup) "<b>" else "") + header.name + ": " +
+                (if (withMarkup) "</b>" else "") + header.value +
+                if (withMarkup) "<br />" else "\n"
+        } ?: ""
+    }
+
+    fun formatByteCount(bytes: Long, si: Boolean): String {
+        val unit = if (si) SI_MULTIPLE else BASE_TWO_MULTIPLE
+
+        if (bytes < unit) {
+            return "$bytes B"
+        }
+
+        val exp = (ln(bytes.toDouble()) / ln(unit.toDouble())).toInt()
+        val pre = (if (si) "kMGTPE" else "KMGTPE")[exp - 1] + if (si) "" else "i"
+
+        return String.format(Locale.US, "%.1f %sB", bytes / unit.toDouble().pow(exp.toDouble()), pre)
+    }
+
+    fun formatJson(json: String): String {
+        return try {
+            val je = JsonParser().parse(json)
+            JsonConverter.instance.toJson(je)
+        } catch (e: JsonParseException) {
+            json
+        }
+    }
+
+    fun formatXml(xml: String): String {
+        return try {
+            val transformerFactory = (SAXTransformerFactory.newInstance() as SAXTransformerFactory).apply {
+                setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true)
+                setAttribute("http://javax.xml.XMLConstants/property/accessExternalDTD", "")
+                setAttribute("http://javax.xml.XMLConstants/property/accessExternalStylesheet", "")
+            }
+
+            val serializer = transformerFactory.newTransformer().apply {
+                setOutputProperty(OutputKeys.INDENT, "yes")
+                setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2")
+            }
+
+            val xmlSource = SAXSource(InputSource(ByteArrayInputStream(xml.toByteArray())))
+            val res = StreamResult(ByteArrayOutputStream())
+            serializer.transform(xmlSource, res)
+            ((res.outputStream as ByteArrayOutputStream).toByteArray()).toString()
+        } catch (e: TransformerException) {
+            xml
+        }
+    }
+
+    @JvmStatic
+    fun getShareText(context: Context, transaction: HttpTransaction): String {
+        var text = ""
```

Comment:
Could be cool if it was kotlinified even more and replaced raw string and the """ notation.